### PR TITLE
add doc about destroying resources and ordering

### PR DESCRIPTION
**Pull request by jbardin**
This adds some documentation and visual examples for resource creation and destruction ordering. The dot text for the graphs is added as comments here for easier updates later on.

### CODE REVIEW

File: docs/destroying.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    b -> a_d [style=dotted];
+    b -> b_d [dir=back];
+}
+-->
+
+In order to resolve these cycles, and all resources that precede a resource
+with `create_before_destroy` must in turn be handled in the same manner.
+Reversing the incoming edged to `A destroy` resolves the problem:
+
+![Correct create_before_destroy replacement](./images/replace_cbd_all.png)
```
Comment (alisdair) [inline docs/destroying.md:262]:
```suggestion
![Correct create_before_destroy replacement](./images/replace_all_cbd.png)
```

File: docs/destroying.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        a_d -> b_d;
+    }
+
+    a -> a_d [dir=back];
+    b -> a_d [dir=back];
+    b -> b_d [dir=back];
+}
+-->
+
+This also demonstrates why `create_before_destry` cannot be overridden when
```
Comment (alisdair) [inline docs/destroying.md:284]:
```suggestion
This also demonstrates why `create_before_destroy` cannot be overridden when
```

File: docs/destroying.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    subgraph nodes {
+        rank=same;
+        a [label="A create"];
+        b [label="B create"];
+        c [label="C create"];
+        b -> c [dir=back];
+        a -> b [dir=back];
+    }
+}
+-->
```
Comment (alisdair) [inline docs/destroying.md:33]: These inline dot graphs in comments are a great idea 👏

**Comment by jbardin** [comment]
Thanks @alisdair, that is the correct order of operations for that graph.

I like the idea of adding a simple list of operations too, since the goal here was to make this as understandable as possible. I think that can wait for a follow up PR, since there will likely be more to add and cleanup here.

**Pull request merged**